### PR TITLE
Add support for Jitsi screensharing in electron app

### PR DESCRIPTION
**Pull request by superdump**

### CODE REVIEW

File: src/components/views/elements/AppTile.js [hunk 1]
```diff
@@ -127,6 +128,30 @@ export default React.createClass({
...
+        }
+
+        if (this.state.widgetUrl.indexOf(event.origin) === -1) {
+            return;
+        }
+
+        if (event.data.widgetAction === 'jitsi_iframe_loaded') {
+            const iframe = this.refs.appFrame.contentWindow
+                .document.querySelector('iframe[id^="jitsiConferenceFrame"]');
+            PlatformPeg.get().setupScreenSharingForIframe(iframe);
```
Comment (t3chguy) [inline src/components/views/elements/AppTile.js]: this method should exist in all platforms, so impl in BasePlatform as a no-op so other platforms inherit it

File: src/components/views/elements/AppTile.js [hunk 2]
```diff
@@ -127,6 +128,30 @@ export default React.createClass({
...
+
+    _onMessage(event) {
+        if (!PlatformPeg.get().isElectron() || this.props.type !== 'jitsi') {
+            return;
+        }
+        if (!event.origin) {
+            event.origin = event.originalEvent.origin;
+        }
+
+        if (this.state.widgetUrl.indexOf(event.origin) === -1) {
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/elements/AppTile.js]: /me points at `this.state.widgetUrl.startsWith(event.origin)`

**Comment by ara4n** [comment]
@superdump please can you pick up the review bits & bobs here so we can merge it? O:-)

File: src/components/views/elements/AppTile.js (hunk 1 above)
Comment (superdump) [reply src/components/views/elements/AppTile.js]: @dbkr - which is preferred? Implement the method in all platforms and just leave it as a stub in every platform other than electron? I would guess so. Alternatively checking if the function exists and is a function could be a compromise...?

File: src/components/views/elements/AppTile.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (superdump) [reply src/components/views/elements/AppTile.js]: Done.

File: src/components/views/elements/AppTile.js (hunk 1 above)
Comment (superdump) [reply src/components/views/elements/AppTile.js]: @dbkr says the former.

**Pull request merged**